Exception on scheduled elastic flush will not schedule it again

**Describe the bug**

On chaos cluster the elastic master was killed by k8s and it happened that the broker was not able to flush the latest events, which is expected. If this happens on exporting this is OK, since it will be retried. If this happens on a scheduled flush this is problematic since we not re-schedule the flush again.

See the following code snippet:
```java
  private void flushAndReschedule() {
    flush();
    scheduleDelayedFlush();
  }
```

**Expected behavior**
Either catch the exception or schedule the flush before calling flush.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
09:44:19.203 [exporter] [zeebe-1.zeebe.default.svc.cluster.local:26501-zb-fs-workers-1] ERROR io.zeebe.util.actor - Actor failed in phase 'STARTED'. Continue with next job.
io.zeebe.exporter.ElasticsearchExporterException: Failed to flush bulk
        at io.zeebe.exporter.ElasticsearchClient.flush(ElasticsearchClient.java:90) ~[zeebe-elasticsearch-exporter-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.exporter.ElasticsearchExporter.flush(ElasticsearchExporter.java:93) ~[zeebe-elasticsearch-exporter-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.exporter.ElasticsearchExporter.flushAndReschedule(ElasticsearchExporter.java:84) ~[zeebe-elasticsearch-exporter-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:76) ~[zeebe-util-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:127) [zeebe-util-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.util.sched.ActorThread.run(ActorThread.java:195) [zeebe-util-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-1 [ACTIVE]
        at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:944) ~[elasticsearch-rest-client-6.8.2.jar:6.8.2]
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:233) ~[elasticsearch-rest-client-6.8.2.jar:6.8.2]
        at org.elasticsearch.client.RestHighLevelClient.internalPerformRequest(RestHighLevelClient.java:1764) ~[elasticsearch-rest-high-level-client-6.8.2.jar:6.8.2]
        at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:1734) ~[elasticsearch-rest-high-level-client-6.8.2.jar:6.8.2]
        at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:1696) ~[elasticsearch-rest-high-level-client-6.8.2.jar:6.8.2]
        at org.elasticsearch.client.RestHighLevelClient.bulk(RestHighLevelClient.java:472) ~[elasticsearch-rest-high-level-client-6.8.2.jar:6.8.2]
        at io.zeebe.exporter.ElasticsearchClient.exportBulk(ElasticsearchClient.java:104) ~[zeebe-elasticsearch-exporter-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        at io.zeebe.exporter.ElasticsearchClient.flush(ElasticsearchClient.java:87) ~[zeebe-elasticsearch-exporter-0.21.0-SNAPSHOT.jar:0.21.0-SNAPSHOT]
        ... 8 more
Caused by: java.net.SocketTimeoutException: 30,000 milliseconds timeout on connection http-outgoing-1 [ACTIVE]
        at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:92) ~[httpasyncclient-4.1.4.jar:4.1.4]
        at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:39) ~[httpasyncclient-4.1.4.jar:4.1.4]
        at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:263) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:492) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:213) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591) ~[httpcore-nio-4.4.11.jar:4.4.11]
        at java.lang.Thread.run(Unknown Source) ~[?:?]
```

</p>
</details>


**Environment:**
 - OS: [e.g. Linux]
 - Zeebe Version: [e.g. 0.20.0]
 - Configuration: [e.g. exporters etc.]

--- a/exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporter.java
+++ b/exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporter.java
@@ -90,10 +90,14 @@ public class ElasticsearchExporter implements Exporter {
   }
 
   private void flush() {
-    if (client.flush()) {
-      controller.updateLastExportedRecordPosition(lastPosition);
-    } else {
-      log.warn("Failed to flush bulk completely");
+    try {
+      if (client.flush()) {
+        controller.updateLastExportedRecordPosition(lastPosition);
+      } else {
+        log.warn("Failed to flush bulk completely");
+      }
+    } catch (Exception e) {
+      log.error("Failed to flush bulk", e);
     }
   }
 

--- a/exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchExporterTest.java
+++ b/exporters/elasticsearch-exporter/src/test/java/io/zeebe/exporter/ElasticsearchExporterTest.java
@@ -296,6 +296,22 @@ public class ElasticsearchExporterTest {
     assertThat(testHarness.getController().getPosition()).isEqualTo(exported.get(3).getPosition());
   }
 
+  @Test
+  public void shouldHandleExceptionOnFlush() {
+    // given
+    when(esClient.shouldFlush()).thenReturn(true);
+    when(esClient.flush()).thenThrow(new ElasticsearchExporterException("expected"));
+
+    createAndOpenExporter();
+
+    // when
+    testHarness.export();
+    testHarness.export();
+
+    // then
+    verify(esClient, times(2)).flush();
+  }
+
   private ElasticsearchExporter createExporter() {
     return createExporter(esClient);
   }

